File: solutions/javascript/SimplifyPath/SimplifyPath.js
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function(path) {
    let arr = path.split('/').filter(x=>x!='');
    let res = [];
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] == '.') {
            continue;
        } else if (arr[i] == '..') {
            if (res.length > 0) {
                res.pop();
            }
        } else {
            res.push(arr[i]);
        }
    }
    return '/' + res.join('/');
};

File: solutions/java/SimplifyPath/SimplifyPath.java
class Solution {
    public String simplifyPath(String path) {
        String[] strings = path.split("/");
        Stack<String> stack = new Stack<>();
        for (String s : strings) {
            if ("".equals(s) || ".".equals(s)) {
                continue;
            } else if ("..".equals(s)) {
                if (!stack.isEmpty()) {
                    stack.pop();
                }
            } else {
                stack.push(s);
            }
        }
        if (stack.isEmpty()) {
            return "/";
        }
        StringBuilder result = new StringBuilder("/");
        for (Object o : stack.toArray()) {
            result.append(o.toString()).append("/");
        }
        return result.toString().substring(0, result.length() - 1);
    }
}